src/types/json.h

```diff
@@ -132,6 +132,44 @@ struct JsonValue {
     return Status::OK();
   }
 
+  Status Type(std::string_view path, std::vector<std::string> *types) const {
+    types->clear();
+    try {
+      jsoncons::jsonpath::json_query(value, path, [&types](const std::string & /*path*/, const jsoncons::json &val) {
+        switch (val.type()) {
+          case jsoncons::json_type::null_value:
+            types->emplace_back("null");
+            break;
+          case jsoncons::json_type::bool_value:
+            types->emplace_back("boolean");
+            break;
+          case jsoncons::json_type::int64_value:
+          case jsoncons::json_type::uint64_value:
+            types->emplace_back("integer");
+            break;
+          case jsoncons::json_type::double_value:
+            types->emplace_back("number");
```

Comment:
Should we support `half` here?

Reply:
ditto https://github.com/apache/kvrocks/pull/1847#discussion_r1367741127